### PR TITLE
Feat: Implement CHES email notification

### PR DESCRIPTION
**Pull request by prv-proton**
#1148 
image

secrets can be found here.
[lcfs-email-service-dev](https://teams.microsoft.com/l/message/19:1fc88441-3dc0-4f50-a6e9-80caec052de5_6f100c7e-b1ed-4e23-8db2-ddc1ff45ef68@unq.gbl.spaces/1732745247246?context=%7B%22contextType%22%3A%22chat%22%7D)

note: view component is only for testing.

### CODE REVIEW

File: backend/lcfs/web/api/email/repo.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            .filter(
+                NotificationType.name == NotificationTypeEnum[notification_type.upper()]
+            )
+            .limit(1)  # Fetch only one record
+        )
+
+        # Execute the query
+        result = await self.db.execute(query)
+        template = result.scalar_one_or_none()
+        return template or "default.html"
```
Comment (dhaselhan) [inline backend/lcfs/web/api/email/repo.py:71]: question: Is this temporary just for testing until we get some templates implemented?

File: backend/lcfs/web/api/email/services.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    Service layer for sending email notifications via CHES.
+    Handles CHES-specific logic and coordinates with the repository.
+    """
+
+    def __init__(self, repo: CHESEmailRepository = Depends()):
+        self.repo = repo
+
+        # CHES configuration
+        self.config = {
+            "AUTH_URL": os.getenv("CHES_AUTH_URL", ""),
```
Comment (dhaselhan) [inline backend/lcfs/web/api/email/services.py:28]: Currently all of our ENV vars start with LCFS_ and are mapped into settings.py, should we follow that pattern here?

File: backend/lcfs/web/api/email/repo.py (hunk 1 above)
Comment (dhaselhan) [reply backend/lcfs/web/api/email/repo.py:71]: ah nvm, I see how this works. We have a generic template for some notifications but some have a specific template.

File: backend/lcfs/web/api/email/services.py (hunk 2 above)
Comment (dhaselhan) [reply backend/lcfs/web/api/email/services.py:28]: Ah nvm, I see how this works. We have a generic default template for some notifications but others have specific content.

**Review by dhaselhan — CHANGES_REQUESTED** [review]
Just want to make sure we are good with a new ENV var pattern.

File: backend/lcfs/web/api/email/services.py (hunk 2 above)
Comment (prv-proton) [reply backend/lcfs/web/api/email/services.py]: Thanks for your review, I've made the necessary changes as per your suggestions.

**Comment by prv-proton** [comment]
> Just want to make sure we are good with a new ENV var pattern.

Thanks @dhaselhan , as per your suggestion modified the code, please review it.

**Review by dhaselhan — CHANGES_REQUESTED** [review]
Please fix the backend tests

**Review by dhaselhan — APPROVED** [review]
LGTM 👍

**Pull request merged**